### ISSUE TITLE
Vernemq crash when the container restart in swarm mode

### ISSUE THREAD

**Issue by sguilly** [vernemq/docker-vernemq#181]
INFRA : docker swarm / 3 nodes

If i don't use DOCKER_VERNEMQ_NODENAME, the vm.args use the IP of the container and the cluster start well.
-> But if the container restart, the container crash (because the IP change ?)

If i set DOCKER_VERNEMQ_NODENAME in my swarm stack, the container restart well but vmq-admin don't work and the cluster don't work.

e2f8a10:~$ vmq-admin cluster show
Node 'VerneMQ@vernemq-1' not responding to pings.

Any tips ?

**Comment by ioolkos** [comment]
Have you reviewed https://github.com/vernemq/docker-vernemq#5-using-docker-swarm ?
Other than that I think I have currently no tip.

**Comment by sguilly** [comment]
Yes, without persistence, i have no issues with the basic example.

But i am able not to find a stable config to repair cluster after a container restarting

My stack file
`version: '3.7'

services:

  vernemq-1:
    image: vernemq/vernemq
    
    ports:
            - target: 1883
              published: 31883
              protocol: tcp
              mode: host
              
            - target: 8883
              published: 38883
              protocol: tcp
              mode: host
              
            - target: 8081
              published: 38081
              protocol: tcp
              mode: host
            
            - target: 8888
              published: 38888
              protocol: tcp
              mode: host

    networks:
      - proxy_proxy

    deploy:
      replicas: 1
      placement:
        constraints: [node.hostname == mqtt-1-opc]

    environment:
      DOCKER_VERNEMQ_SWARM: 1
      DOCKER_VERNEMQ_NODENAME: vernemq-1

    volumes:
      - vernemq-1-data:/vernemq/data

    configs:
      - source: vernemq.conf.v15
        target: /vernemq/etc/vernemq.conf.local

    secrets:
      - source: ssl_server_cert
        target: /ssl/cert.pem
      - source: ssl_server_key
        target: /ssl/privkey.pem
        
  vernemq-2:
    image: vernemq/vernemq
    
    depends_on:
      - vernemq-1
    
    ports:
            - target: 1883
              published: 31883
              protocol: tcp
              mode: host
              
            - target: 8883
              published: 38883
              protocol: tcp
              mode: host
              
            - target: 8081
              published: 38081
              protocol: tcp
              mode: host

    networks:
      - proxy_proxy

    deploy:
      replicas: 1
      placement:
        constraints: [node.hostname == mqtt-2-opc]

    environment:
      DOCKER_VERNEMQ_SWARM: 1
      DOCKER_VERNEMQ_DISCOVERY_NODE: vernemq-1
      DOCKER_VERNEMQ_NODENAME: vernemq-2
      
    volumes:
      - vernemq-2-data:/vernemq/data

    configs:
      - source: vernemq.conf.v15
        target: /vernemq/etc/vernemq.conf.local

    secrets:
      - source: ssl_server_cert
        target: /ssl/cert.pem
      - source: ssl_server_key
        target: /ssl/privkey.pem 
        
  vernemq-3:
    image: vernemq/vernemq
    
    depends_on:
      - vernemq-1

    ports:
            - target: 1883
              published: 31883
              protocol: tcp
              mode: host
              
            - target: 8883
              published: 38883
              protocol: tcp
              mode: host
              
            - target: 8081
              published: 38081
              protocol: tcp
              mode: host

    networks:
      - proxy_proxy

    deploy:
      replicas: 1
      placement:
        constraints: [node.hostname == mqtt-3-opc]

    environment:
      DOCKER_VERNEMQ_SWARM: 1
      DOCKER_VERNEMQ_DISCOVERY_NODE: vernemq-1
      DOCKER_VERNEMQ_NODENAME: vernemq-3

    volumes:
      - vernemq-3-data:/vernemq/data

    configs:
      - source: vernemq.conf.v15
        target: /vernemq/etc/vernemq.conf.local

    secrets:
      - source: ssl_server_cert
        target: /ssl/cert.pem
      - source: ssl_server_key
        target: /ssl/privkey.pem
       

networks:
  proxy_proxy:
    external: true

volumes:
  vernemq-1-data:
  vernemq-2-data:
  vernemq-3-data:

configs:
  vernemq.conf.v15:
    external: true

secrets:
  ssl_server_cert:
    external: true
  ssl_server_key:
    external: true`

My conf file
`


accept_eula = yes

listener.tcp.default = 0.0.0.0:1883
listener.ssl.default = 0.0.0.0:8883
listener.ws.default = 0.0.0.0:8080
listener.wss.default = 0.0.0.0:8081
listener.http.default = 0.0.0.0:8888
listener.vmq.clustering = ###IPADDRESS###:44053

nodename = VerneMQ@###IPADDRESS###

allow_anonymous = off

allow_register_during_netsplit = on
allow_publish_during_netsplit = on
allow_subscribe_during_netsplit = on
allow_unsubscribe_during_netsplit = on

allow_multiple_sessions = on

retry_interval = 20

max_client_id_size = 100

persistent_client_expiration = 1w

max_inflight_messages = 20

max_online_messages = -1

max_offline_messages = -1

max_message_size = 0

upgrade_outgoing_qos = off

listener.max_connections = 10000


listener.nr_of_acceptors = 10



listener.mountpoint = off

listener.ssl.certfile = /ssl/cert.pem
listener.wss.certfile = /ssl/cert.pem

listener.ssl.keyfile = /ssl/privkey.pem
listener.wss.keyfile = /ssl/privkey.pem

listener.ssl.tls_version = tlsv1.2

listener.https.tls_version = tlsv1.2

systree_enabled = on

systree_interval = 20000

shared_subscription_policy = prefer_local


plugins.vmq_passwd = off

plugins.vmq_acl = off

plugins.vmq_diversity = off

plugins.vmq_webhooks = on

plugins.vmq_bridge = off

vmq_diversity.script_dir = /usr/share/vernemq/lua

vmq_diversity.auth_mysql.enabled = off

vmq_diversity.auth_mongodb.enabled = off

vmq_diversity.auth_redis.enabled = off

vmq_webhooks.mywebhook1.hook = auth_on_register
vmq_webhooks.mywebhook1.endpoint = http://vernemq-api:3000/auth/register

vmq_webhooks.mywebhook2.hook = auth_on_subscribe
vmq_webhooks.mywebhook2.endpoint = http://vernemq-api:3000/auth/subscribe

vmq_webhooks.mywebhook3.hook = auth_on_subscribe_m5
vmq_webhooks.mywebhook3.endpoint = http://vernemq-api:3000/auth/subscribe

vmq_webhooks.mywebhook4.hook = auth_on_publish
vmq_webhooks.mywebhook4.endpoint = http://vernemq-api:3000/auth/publish

log.console = console

log.console.level = info

log.console.file = /vernemq/etc/console.log

log.error.file = /vernemq/etc/error.log

log.syslog = off

log.crash = on

log.crash.file = /var/log/vernemq/crash.log

log.crash.maximum_message_size = 64KB

log.crash.size = 10MB

log.crash.rotation = $D0

log.crash.rotation.keep = 5

distributed_cookie = vmq_scififarms

erlang.async_threads = 64

erlang.max_ports = 262144

leveldb.maximum_memory.percent = 70`

**Comment by Qanah** [comment]
@sguilly any update? please help me if you fix this issue

**Comment by sguilly** [comment]
@Qanah : i don't find a fix. 

-> Install directly vernemq on the host OS.
